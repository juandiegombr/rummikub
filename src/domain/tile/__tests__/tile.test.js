const TileService = require('../../../domain/tile')

it('should generate all the tiles', async () => {
  const tiles =  TileService.generateTiles()

  expect(tiles).toHaveLength(52)
  expect(tiles).toMatchSnapshot()
})

it('should shuffle the tiles', async () => {
  const tiles =  TileService.generateTiles()
  const shuffledTiles =  TileService.shuffle(tiles)

  expect(shuffledTiles).toHaveLength(52)
})