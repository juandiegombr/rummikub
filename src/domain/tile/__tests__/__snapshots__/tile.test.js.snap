// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generates all the tiles 1`] = `
Array [
  Object {
    "code": "id_red_1",
    "color": "red",
    "value": 1,
  },
  Object {
    "code": "id_red_2",
    "color": "red",
    "value": 2,
  },
  Object {
    "code": "id_red_3",
    "color": "red",
    "value": 3,
  },
  Object {
    "code": "id_red_4",
    "color": "red",
    "value": 4,
  },
  Object {
    "code": "id_red_5",
    "color": "red",
    "value": 5,
  },
  Object {
    "code": "id_red_6",
    "color": "red",
    "value": 6,
  },
  Object {
    "code": "id_red_7",
    "color": "red",
    "value": 7,
  },
  Object {
    "code": "id_red_8",
    "color": "red",
    "value": 8,
  },
  Object {
    "code": "id_red_9",
    "color": "red",
    "value": 9,
  },
  Object {
    "code": "id_red_10",
    "color": "red",
    "value": 10,
  },
  Object {
    "code": "id_red_11",
    "color": "red",
    "value": 11,
  },
  Object {
    "code": "id_red_12",
    "color": "red",
    "value": 12,
  },
  Object {
    "code": "id_red_13",
    "color": "red",
    "value": 13,
  },
  Object {
    "code": "id_blue_1",
    "color": "blue",
    "value": 1,
  },
  Object {
    "code": "id_blue_2",
    "color": "blue",
    "value": 2,
  },
  Object {
    "code": "id_blue_3",
    "color": "blue",
    "value": 3,
  },
  Object {
    "code": "id_blue_4",
    "color": "blue",
    "value": 4,
  },
  Object {
    "code": "id_blue_5",
    "color": "blue",
    "value": 5,
  },
  Object {
    "code": "id_blue_6",
    "color": "blue",
    "value": 6,
  },
  Object {
    "code": "id_blue_7",
    "color": "blue",
    "value": 7,
  },
  Object {
    "code": "id_blue_8",
    "color": "blue",
    "value": 8,
  },
  Object {
    "code": "id_blue_9",
    "color": "blue",
    "value": 9,
  },
  Object {
    "code": "id_blue_10",
    "color": "blue",
    "value": 10,
  },
  Object {
    "code": "id_blue_11",
    "color": "blue",
    "value": 11,
  },
  Object {
    "code": "id_blue_12",
    "color": "blue",
    "value": 12,
  },
  Object {
    "code": "id_blue_13",
    "color": "blue",
    "value": 13,
  },
  Object {
    "code": "id_orange_1",
    "color": "orange",
    "value": 1,
  },
  Object {
    "code": "id_orange_2",
    "color": "orange",
    "value": 2,
  },
  Object {
    "code": "id_orange_3",
    "color": "orange",
    "value": 3,
  },
  Object {
    "code": "id_orange_4",
    "color": "orange",
    "value": 4,
  },
  Object {
    "code": "id_orange_5",
    "color": "orange",
    "value": 5,
  },
  Object {
    "code": "id_orange_6",
    "color": "orange",
    "value": 6,
  },
  Object {
    "code": "id_orange_7",
    "color": "orange",
    "value": 7,
  },
  Object {
    "code": "id_orange_8",
    "color": "orange",
    "value": 8,
  },
  Object {
    "code": "id_orange_9",
    "color": "orange",
    "value": 9,
  },
  Object {
    "code": "id_orange_10",
    "color": "orange",
    "value": 10,
  },
  Object {
    "code": "id_orange_11",
    "color": "orange",
    "value": 11,
  },
  Object {
    "code": "id_orange_12",
    "color": "orange",
    "value": 12,
  },
  Object {
    "code": "id_orange_13",
    "color": "orange",
    "value": 13,
  },
  Object {
    "code": "id_black_1",
    "color": "black",
    "value": 1,
  },
  Object {
    "code": "id_black_2",
    "color": "black",
    "value": 2,
  },
  Object {
    "code": "id_black_3",
    "color": "black",
    "value": 3,
  },
  Object {
    "code": "id_black_4",
    "color": "black",
    "value": 4,
  },
  Object {
    "code": "id_black_5",
    "color": "black",
    "value": 5,
  },
  Object {
    "code": "id_black_6",
    "color": "black",
    "value": 6,
  },
  Object {
    "code": "id_black_7",
    "color": "black",
    "value": 7,
  },
  Object {
    "code": "id_black_8",
    "color": "black",
    "value": 8,
  },
  Object {
    "code": "id_black_9",
    "color": "black",
    "value": 9,
  },
  Object {
    "code": "id_black_10",
    "color": "black",
    "value": 10,
  },
  Object {
    "code": "id_black_11",
    "color": "black",
    "value": 11,
  },
  Object {
    "code": "id_black_12",
    "color": "black",
    "value": 12,
  },
  Object {
    "code": "id_black_13",
    "color": "black",
    "value": 13,
  },
]
`;
